src/app/InteractionModelEngine.h

```diff
@@ -411,6 +415,9 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
     // Returns the old data model provider value.
     DataModel::Provider * SetDataModelProvider(DataModel::Provider * model);
 
+    /* DataModel::ActionContext implementation */
+    Messaging::ExchangeContext * CurrentExchange() override { return mCurrentExchange; }
```

Comment:
Can we make this private, please?  This API is not supposed to be used on InteractionModelEngine directly.

Reply:
Moved the ActionContext as private inheritance. DM providers will see it because that is the best alternative to using actual globals, however I agree it is odd. This is because we have to live with code that exists today.